### PR TITLE
(@wdio/appium-service): Delete error log on success executions

### PR DESCRIPTION
**Pull request by MadSandwich**
## Proposed changes

Fixes logging into console error even is run was successful. A small patch on https://github.com/webdriverio/webdriverio/pull/11176

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-appium-service/src/launcher.ts [hunk 1]
```diff
@@ -173,7 +173,6 @@ export default class AppiumLauncher implements Services.ServiceInstance {
             } else if (error) {
                 errorMessage += `\n${error.toString()}`
             }
-            log.error(errorMessage)
```
Comment (christian-bromann) [inline packages/wdio-appium-service/src/launcher.ts:176]: I think we should not remove it but instead check if error code is !== 0 and only log then

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: packages/wdio-appium-service/src/launcher.ts (hunk 1 above)
Comment (MadSandwich) [reply packages/wdio-appium-service/src/launcher.ts:176]: sure, make sense.

File: packages/wdio-appium-service/src/launcher.ts [hunk 2]
```diff
@@ -173,8 +173,8 @@ export default class AppiumLauncher implements Services.ServiceInstance {
...
             callback(new Error(errorMessage), null)
+            exitCode !== 0 ? log.error(errorMessage) : ''
```
Comment (christian-bromann) [inline packages/wdio-appium-service/src/launcher.ts:177]: Let's log this before the callback is called

Comment (christian-bromann) [reply packages/wdio-appium-service/src/launcher.ts:177]: and do:
```ts
if (exitCode !== 0) {
    log.error(errorMessage)
}
```

Comment (MadSandwich) [reply packages/wdio-appium-service/src/launcher.ts]: rewritten

**Pull request merged**